In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [60]:
tivio_data = pd.read_excel("C:/Users/sadma/Downloads/Tivo Survey Case Data.xlsx")
tivio_data

ID  Gender Marital Status   Work Status Education  \
0       1    male        married  professional      none   
1       2    male         single          none      none   
2       3    male        married  professional        BA   
3       4    male        married          none       PhD   
4       5  female         single          none      none   
..    ...     ...            ...           ...       ...   
995   996  female         single  professional      none   
996   997    male        married  professional        MA   
997   998    male        married  professional      none   
998   999    male        married          none        BA   
999  1000  female        married  professional      none   

     Annual Income (x1000 $)  Age       Location Purchasing Decision-maker  \
0                         49   30        Florida                    family   
1                         46   36        Alabama                    single   
2                         58   66  Massachusetts                    family   
3                         51   78       New York                    family   
4                         46   52        Montana                    single   
..                       ...  ...            ...                       ...   
995                       29   51        Florida                    family   
996                       32   73          Maine                    single   
997                       28   65        Georgia                    single   
998                       49   67          Maine                    family   
999                       33   58      Tennessee                    single   

           Purchasing Location  Monthly Electronics Spend  \
0    mass-consumer electronics                         35   
1    mass-consumer electronics                         35   
2             specialty stores                         64   
3    mass-consumer electronics                         33   
4    mass-consumer electronics                         45   
..                         ...                        ...   
995                     retail                         15   
996                     retail                         15   
997                   discount                         10   
998  mass-consumer electronics                         41   
999                     retail                         23   

     Monthly Household Spend  Purchasing Frequency (every x months)  \
0                        150                                     13   
1                        163                                     26   
2                        103                                     13   
3                        154                                     22   
4                        161                                     47   
..                       ...                                    ...   
995                       54                                     21   
996                       73                                     21   
997                       29                                     11   
998                       55                                     23   
999                      104                                      1   

    Technology Adoption  TV Viewing (hours/day)  \
0                  late                       2   
1                  late                      10   
2                 early                       0   
3                  late                       5   
4                  late                       2   
..                  ...                     ...   
995               early                       2   
996               early                       0   
997               early                       1   
998                late                      10   
999               early                       1   

                               Favorite feature  
0    saving favorite shows to watch as a family  
1    saving favorite shows to watch as a family  


## 1. When working with data in a business context it is important to be able to identify the type of data you are working with in relation to the business problem. Please categorize each data field by its segmentation parameter type: “Demographic,” “Purchasing,” “Behavioral,” “Attitude,” or “Other.”

**"Categorizing Data Fields by Segmentation Parameter Type"**
 
<b>Demographic:

* ID
* Gender
* Marital Status
* Education
* Annual Income (x1000 $)
* Age
* Location

<b>Purchasing:

* Purchasing Decision-maker
* Purchasing Location
* Monthly Electronics Spend
* Monthly Household Spend
* Purchasing Frequency (every x months)

<b>Behavioral:

* Technology Adoption
* TV Viewing (hours/day)

<b>Attitude:

* Favorite feature

## 2. In the Survey Data worksheet, do the following:

a. Insert a column labeled “Annual spending on electronics” that calculates the
annual spending on electronics.

b. Insert another column labeled “Spending as % of income” that calculates annual spending on electronics as a percentage of annual income.


### a. Insert a column labeled “Annual spending on electronics” that calculates the annual spending on electronics.

In [61]:
tivio_data['Annual spending on electronics'] = tivio_data['Monthly Electronics Spend'] * 12

### b. Insert another column labeled “Spending as % of income” that calculates annual spending on electronics as a percentage of annual income.

In [62]:
# Convert the 'Annual spending columns to represnt the data values in 1000s'
tivio_data["Spending as % of income"] = (tivio_data['Annual spending on electronics'] / (tivio_data['Annual Income (x1000 $)']*1000)) * 100
tivio_data[['Annual spending on electronics', 'Spending as % of income']]

Annual spending on electronics  Spending as % of income
0                               420                 0.857143
1                               420                 0.913043
2                               768                 1.324138
3                               396                 0.776471
4                               540                 1.173913
..                              ...                      ...
995                             180                 0.620690
996                             180                 0.562500
997                             120                 0.428571
998                             492                 1.004082
999                             276                 0.836364

[1000 rows x 2 columns]

## 3. Create a table for each attribute (e.g., Gender) and record the percentage of responses for each answer (e.g., % of respondents answering "male" and % answering “female”).

In [63]:
attributes = ['Gender', 'Marital Status', 'Work Status', 'Education', 'Location', 
                           'Purchasing Decision-maker', 'Purchasing Location', 'Technology Adoption', 'Favorite feature']
attribute_percentages = {}
for attribute in attributes:
    attribute_percentages[attribute] = (tivio_data[attribute].value_counts(normalize=True) * 100).reset_index().rename(columns={'index': attribute, attribute: 'Percentage'})

In [64]:
table_attributes = ['Gender', 'Marital Status', 'Work Status', 'Education', 'Location', 
                    'Purchasing Decision-maker', 'Purchasing Location', 'Technology Adoption', 'Favorite feature']

table_attribute_percentages = {}

for attribute in table_attributes:
    count = tivio_data[attribute].value_counts(normalize=True) * 100
    table_attribute_percentages[attribute] = pd.DataFrame({'Percentage': counts}).reset_index().rename(columns={'index': attribute})


In [65]:
attribute_percentages['Marital Status']

Percentage  proportion
0    married        72.0
1     single        28.0

In [66]:
attribute_percentages["Gender"]

Percentage  proportion
0       male        53.5
1     female        46.5

In [67]:
attribute_percentages['Work Status']

Percentage  proportion
0  professional        65.0
1          none        35.0

In [68]:
tivio_data.corr(numeric_only=True)

ID  Annual Income (x1000 $)  \
ID                                     1.000000                 0.010652   
Annual Income (x1000 $)                0.010652                 1.000000   
Age                                   -0.041543                 0.132465   
Monthly Electronics Spend             -0.027364                 0.361016   
Monthly Household Spend               -0.054874                 0.257187   
Purchasing Frequency (every x months)  0.024479                -0.126362   
TV Viewing (hours/day)                -0.010367                 0.084335   
Annual spending on electronics        -0.027364                 0.361016   
Spending as % of income               -0.008861                 0.040278   

                                            Age  Monthly Electronics Spend  \
ID                                    -0.041543                  -0.027364   
Annual Income (x1000 $)                0.132465                   0.361016   
Age                                    1.000000                  -0.044861   
Monthly Electronics Spend             -0.044861                   1.000000   
Monthly Household Spend               -0.012586                   0.803806   
Purchasing Frequency (every x months) -0.001293                  -0.277886   
TV Viewing (hours/day)                -0.035950                   0.247169   
Annual spending on electronics        -0.044861                   1.000000   
Spending as % of income               -0.302745                   0.819872   

                                       Monthly Household Spend  \
ID                                                   -0.054874   
Annual Income (x1000 $)                               0.257187   
Age                                                  -0.012586   
Monthly Electronics Spend                             0.803806   
Monthly Household Spend                               1.000000   
Purchasing Frequency (every x months)                -0.233818   
TV Viewing (hours/day)                                0.191617   
Annual spending on electronics                        0.803806   
Spending as % of income                               0.651908   

                                       Purchasing Frequency (every x months)  \
ID                                                                  0.024479   
Annual Income (x1000 $)                                            -0.126362   
Age                                                                -0.001293   
Monthly Electronics Spend                                          -0.277886   
Monthly Household Spend                                            -0.233818   
Purchasing Frequency (every x months)                               1.000000   
TV Viewing (hours/day)                                              0.163497   
Annual spending on electronics                                     -0.277886   
Spending as % of income                                            -0.183438   

                                       TV Viewing (hours/day)  \
ID                                                  -0.010367   
Annual Income (x1000 $)                              0.084335   
Age                                                 -0.035950   
Monthly Electronics Spend                            0.247169   
Monthly Household Spend                              0.191617   
Purchasing Frequency (every x months)                0.163497   
TV Viewing (hours/day)                               1.000000   
Annual spending on electronics                       0.247169   
Spending as % of income                              0.235795   

                                       Annual spending on electronics  \
ID                                                          -0.027364   
Annual Income (x1000 $)                                      0.361016   
Age                                                         -0.044861   
Monthly Electronics Spend                                    1.000000   
Monthly Household Spend  

**a. Age vs. Purchasing Frequency**

* Relationship: Weak negative correlation (-0.00129) indicates a slight tendency that as age increases, there is a very small decrease in purchasing frequency.
* Insight: The correlation is close to zero, suggesting that age is not a significant predictor of purchasing frequency.


**b. Age vs. TV Viewing**

* Relationship: Weak negative correlation (-0.03595) implies a small tendency that as age increases, there is a slight decrease in TV viewing hours per day.
* Insight: The correlation is relatively weak, indicating that age may not be a strong factor influencing TV viewing habits.

**c. Annual Income vs. TV Viewing**
* Relationship: Weak positive correlation (0.08433) suggests a slight tendency that as annual income increases, there is a small increase in TV viewing hours per day.
* Insight: The correlation is weak, indicating that there may be a minor association between higher income and slightly more TV viewing.


**d. Monthly Electronic Spending vs Monthly Household Spending**
* Relationship: Strong positive correlation (0.80381) indicates a substantial positive relationship, suggesting that as monthly electronic spending increases, there is a strong tendency for monthly household spending to also increase.
* Insight: The strong correlation suggests a close association between spending on electronics and overall household spending.



**f. Other Relationships & Redundancy**
* The Annual spending on electronics is rendered redundant, as it mirrors the behavior of Monthly Electronics Spend and as a result the corrlation value is exactly 1

**e. Education vs. Favorite Feature:**

* This is not a numeric value so we will not be able to calauclate the correlation value for this 

## a.	How many married men who are early adopters have monthly electronics spend high enough that they can afford to purchase a TiVo for $499 and still be able to spend more on electronics in the next two years? 

In [93]:
# Filtering out married man who were early adopters

early_adopters = tivio_data[(tivio_data['Gender'] == 'male') & (tivio_data['Marital Status'] == 'married')&(tivio_data['Technology Adoption'] == 'early')]

tivo_cost = 499

affordability_condition = (tivio_data['Monthly Electronics Spend'] * 24) > tivio_cost + (tivio_data['Monthly Electronics Spend'] * 12 * 2)


able_to_purchase_tivo = early_adopters[early_adopters['Monthly Electronics Spend'] * 24 > tivo_cost]
num_married_men_able_to_purchase_tivo = len(able_to_purchase_tivo)
num_married_men_able_to_purchase_tivo

134

In [99]:
tivio_cost = 499

# Filter criteria
filtered_data = tivio_data[(tivio_data['Gender'] == 'male') & (tivio_data['Marital Status'] == 'married')&(tivio_data['Technology Adoption'] == 'early')]

affordability_condition = (tivio_data['Monthly Electronics Spend'] * 24) > tivio_cost + (tivio_data['Monthly Electronics Spend'] * 12 * 2)
                           
early_adopters = filtered_data[filtered_data['Monthly Electronics Spend'] * 24 > tivo_cost]

# Count the number of individuals
count_married_early_adopters = len(early_adopters)


print(f"The number of married men who are early adopters and can afford to purchase a TiVo for $499 and spend more in the next two years is: {count_married_early_adopters}")


The number of married men who are early adopters and can afford to purchase a TiVo for $499 and spend more in the next two years is: 134


## b. How many women with an education level of MA or PhD are making purchasing decisions for electronics without discussing them with a spouse, either because they are single or because they are making purchasing decisions without the involvement of their spouses? 

In [129]:
# Filter criteria
filtered_data_women_ma_phd = tivio_data[(tivio_data['Gender'] == 'female') &
                                        ((tivio_data['Education'] == 'MA') | (tivio_data['Education'] == 'PhD')) &
                                        ((tivio_data['Marital Status'] == 'single') | (tivio_data['Purchasing Decision-maker'] == 'single'))]

count_women_ma_phd_purchasing = len(filtered_data_women_ma_phd)


print(f"The number of women with an education level of MA or PhD making purchasing decisions without discussing them with a spouse is: {count_women_ma_phd_purchasing}")


The number of women with an education level of MA or PhD making purchasing decisions without discussing them with a spouse is: 56


## c. How many early adopters purchase electronics at least once every year, and do so in stores that specialize in electronics? 

In [134]:
filtered_data_specialty_stores = tivio_data[(tivio_data['Technology Adoption'] == 'early') &
                                             (tivio_data['Purchasing Frequency (every x months)'] <= 12) &
                                             (tivio_data['Purchasing Location'] == 'specialty stores')]

count_adopters_specialty_stores = len(filtered_data_specialty_stores)
print(f"The number of  early adopters purchase electronics at least once every year, and do so in stores that specialize in electronics:{count_adopters_specialty_stores}")


The number of  early adopters purchase electronics at least once every year, and do so in stores that specialize in electronics:132


## d. How many seniors (over the age of 65) spend more than six hours a day watching TV? What is their income range? What is their average annual income? 

In [32]:
seniors_watching_tv = tivio_data[(tivio_data['Age'] > 65) & (tivio_data['TV Viewing (hours/day)'] > 6)]
num_seniors_watching_tv = len(seniors_watching_tv)
income_range_seniors_watching_tv = seniors_watching_tv['Annual Income (x1000 $)'].min(), seniors_watching_tv['Annual Income (x1000 $)'].max()
average_income_seniors_watching_tv = seniors_watching_tv['Annual Income (x1000 $)'].mean()
num_seniors_watching_tv, income_range_seniors_watching_tv, average_income_seniors_watching_tv

(20, (41, 55), 48.6)

In [143]:
seniors_wtachingTV_six_hours_a_day = tivio_data[(tivio_data['Age'] > 65) & (tivio_data['TV Viewing (hours/day)'] > 6)]

count_active_seniors = len(seniors_wtachingTV_six_hours_a_day)
seniors_income_range = seniors_wtachingTV_six_hours_a_day['Annual Income (x1000 $)'].min(), seniors_wtachingTV_six_hours_a_day['Annual Income (x1000 $)'].max()
average_income_seniors = seniors_wtachingTV_six_hours_a_day['Annual Income (x1000 $)'].mean()

print(f"Number of active seniors: {num_active_seniors}")
print(f"Income range of seniors: {seniors_income_range}")
print(f"Average income of active seniors: {average_income_seniors}")


Number of active seniors: 20
Income range of seniors: (41, 55)
Average income of active seniors: 48.6


In [160]:
#a. Gender with Monthly Electronic Spend

gender_monthly_electronic_spend_summary = tivio_data.groupby('Gender')['Monthly Electronics Spend'].describe().T

#b. Martial Status with Monthly Electronic Spend
marital_status_monthly_electronic_spend_summary = tivio_data.groupby('Marital Status')['Monthly Electronics Spend'].describe().T

#c. Education with Monthly Electronic Spend
education_monthly_electronic_spend_summary = tivio_data.groupby('Education')['Monthly Electronics Spend'].describe().T


print("a. Gender with Monthly Electronic Spend:")
print(gender_monthly_electronic_spend_summary)

print("\nb. Martial Status with Monthly Electronic Spend:")
print(marital_status_monthly_electronic_spend_summary)

print("\nc. Education with Monthly Electronic Spend:")
print(education_monthly_electronic_spend_summary)


a. Gender with Monthly Electronic Spend:
Gender      female        male
count   465.000000  535.000000
mean     28.776344   32.861682
std      15.120261   18.368460
min       7.000000    9.000000
25%      17.000000   17.000000
50%      23.000000   30.000000
75%      38.000000   45.000000
max      83.000000   88.000000

b. Martial Status with Monthly Electronic Spend:
Marital Status     married      single
count           720.000000  280.000000
mean             30.704167   31.625000
std              17.491424   15.868594
min               9.000000    7.000000
25%              17.000000   18.000000
50%              21.000000   32.000000
75%              44.000000   36.000000
max              85.000000   88.000000

c. Education with Monthly Electronic Spend:
Education          BA          MA         PhD        none
count      260.000000  124.000000  110.000000  506.000000
mean        40.242308   40.370968   38.290909   22.294466
std         18.711149   14.678931   12.102275   12.583749
mi

## 7. Based on your analysis of the dataset, who do you believe the ideal customer group  who marketing should focus their attention on? 


### Ideal Customer Group
    
Based on this analysis, the ideal customer group for marketing focus could be:
    
<b>Income Bracket: Target individuals with an annual income of $40,000 to $60,000, as they have a moderate spending on electronics.
    
<b> Age Group: Focus on individuals in their 30s to 60s, especially considering the slight trend of increased TV viewing with age.
<b> Technology Adoption: Both early and late adopters are significant, but early adopters might be more receptive to new technologies like Tivo.

<b> TV Viewing Habits: Individuals who watch TV for 2 to 5 hours per day could be prime targets, as they spend a substantial amount of time in front of the TV.